x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartTrainedModelDeploymentAction.java

```diff
@@ -150,6 +169,22 @@ protected void masterOperation(Task task, StartTrainedModelDeploymentAction.Requ
         client.execute(GetTrainedModelsAction.INSTANCE, getModelRequest, getModelListener);
     }
 
+    private void getModelBytes(TrainedModelConfig trainedModelConfig, ActionListener<Long> listener) {
+        ChunkedTrainedModelRestorer restorer = new ChunkedTrainedModelRestorer(trainedModelConfig.getLocation().getModelId(),
+            client, threadPool.executor(MachineLearning.UTILITY_THREAD_POOL_NAME), xContentRegistry);
+        restorer.setSearchIndex(trainedModelConfig.getLocation().getResourceName());
+        restorer.setSearchSize(1);
+        restorer.restoreModelDefinition(
+            doc -> {
+                listener.onResponse(doc.getTotalDefinitionLength());
```

Comment:
To me, this is non-obvious. I fail to see how `len(compress_json_object)` is an adequate approximation to the neural net being loaded in memory.

Reply:
This is what I was thinking about in https://github.com/elastic/elasticsearch/pull/75378#discussion_r670456886